src/sig_stfl/lms/sig_stfl_lms_functions.c

```diff
@@ -40,25 +40,50 @@ typedef struct OQS_LMS_KEY_DATA {
 
 	/* secret key data */
 	uint8_t *sec_key;
+
+	/* app specific */
+	void *context;
 } oqs_lms_key_data;
 
 OQS_API OQS_STATUS OQS_SIG_STFL_alg_lms_sign(uint8_t *signature, size_t *signature_length, const uint8_t *message,
         size_t message_len, OQS_SIG_STFL_SECRET_KEY *secret_key) {
 
+	OQS_STATUS status = OQS_ERROR;
+	OQS_STATUS rc_keyupdate = OQS_ERROR;
 	if (secret_key == NULL || message == NULL || signature == NULL) {
-		return OQS_ERROR;
+		return status;
 	}
 
-	/* TODO: Make sure we have a way to update the private key */
-
 	if (oqs_sig_stfl_lms_sign(secret_key, signature,
 	                          signature_length,
 	                          message, message_len) != 0) {
-		return OQS_ERROR;
+		return status;
+	}
+
+	/* Update/increment to the next private key */
+	if (secret_key->secure_store_scrt_key) {
+		oqs_lms_key_data *lms_key_data = (oqs_lms_key_data *)secret_key->secret_key_data;
+		if (lms_key_data) {
+			const OQS_SIG_STFL_SECRET_KEY *sk = secret_key;
+			size_t sk_key_buf_len = 0;
+			uint8_t *sk_key_buf = NULL;
+			void *context = lms_key_data->context;
+
+			rc_keyupdate = oqs_serialize_lms_key(sk, &sk_key_buf_len, &sk_key_buf);
+			if (rc_keyupdate == OQS_SUCCESS) {
+				rc_keyupdate = secret_key->secure_store_scrt_key(sk_key_buf, sk_key_buf_len, context);
+				if (rc_keyupdate == OQS_SUCCESS) {
+					return OQS_SUCCESS;
+				} else {
+					return OQS_ERROR;
+				}
+			}
+			return status;
+		}
+	} else {
+		return status;
 	}
-
-	/* TODO: Update private key */
-	return OQS_SUCCESS;
+	return status;
```

Comment:
Nit: I think you can flatten this out, if there is something wrong, proceed to early return, otherwise continue to the main functionalities.

Reply:
Done.